File: infra/gateway/src/main/java/com/sesac/carematching/infra/gateway/config/Http2UpstreamConfig.java
package com.sesac.carematching.infra.gateway.config;

import org.springframework.cloud.gateway.config.HttpClientCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import reactor.netty.http.HttpProtocol;

@Configuration
public class Http2UpstreamConfig {

    @Bean
    public HttpClientCustomizer http2ClientCustomizer() {
        return httpClient -> httpClient.protocol(HttpProtocol.H2C);
    }
}

File: infra/gateway/src/main/kotlin/com/sesac/carematching/infra/gateway/config/Http2UpstreamConfig.kt
package com.sesac.carematching.infra.gateway.config

import org.springframework.cloud.gateway.config.HttpClientCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import reactor.netty.http.HttpProtocol

@Configuration
class Http2UpstreamConfig {

    @Bean
    fun http2ClientCustomizer(): HttpClientCustomizer =
        HttpClientCustomizer { httpClient -> httpClient.protocol(HttpProtocol.H2C) }
}
